### ISSUE TITLE
Unable to Open Instance Details Page

### ISSUE THREAD

**Issue by charles2588** [apache/airflow#15413]
**Apache Airflow version**: 2.0

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): N/A

**Environment**: 

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release): 
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04
```
- **Kernel** (e.g. `uname -a`):  `Linux airflowvm 5.4.0-70-generic #78-Ubuntu SMP Fri Mar 19 13:29:52 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
- **Install tools**: 
- **Others**:

**What happened**:

  The instance details page throws error:-
http://localhost:8080/task?dag_id=xcom_dag&task_id=downloading_data&execution_date=2021-04-16T23%3A01%3A08.044070%2B00%3A00

![image](https://user-images.githubusercontent.com/5734421/115091805-71d1ef00-9ecd-11eb-9ab1-839d5e8a7e51.png)

`Something bad has happened.
Please consider letting us know by creating a bug report using GitHub.

Python version: 3.8.5
Airflow version: 2.0.0
Node: airflowvm
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/airflow/sandbox/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/airflow/sandbox/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/airflow/sandbox/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/airflow/sandbox/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/airflow/www/auth.py", line 34, in decorated
    return func(*args, **kwargs)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/airflow/www/decorators.py", line 60, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/airflow/www/views.py", line 1193, in task
    attr = getattr(ti, attr_name)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 792, in previous_start_date_success
    return self.get_previous_start_date(state=State.SUCCESS)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/airflow/utils/session.py", line 65, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 776, in get_previous_start_date
    return prev_ti and pendulum.instance(prev_ti.start_date)
  File "/home/airflow/sandbox/lib/python3.8/site-packages/pendulum/__init__.py", line 174, in instance
    raise ValueError("instance() only accepts datetime objects.")
ValueError: instance() only accepts datetime objects.`

**What you expected to happen**:

**How to reproduce it**:

1. Go to Airflow UI and open any  dag you may have
2. Click on any task node in Graph view
3. Click on Instance Details button

![image](https://user-images.githubusercontent.com/5734421/115092500-15bb9a80-9ece-11eb-8e31-3182c8300fe4.png)

![image](https://user-images.githubusercontent.com/5734421/115092769-279d3d80-9ece-11eb-90e8-bed3794826d5.png)

![image](https://user-images.githubusercontent.com/5734421/115093135-43084880-9ece-11eb-8461-0a8ede567eac.png)

Note the Host operating system is windows 10 and Chrome 85

**Anything else we need to know**:

The issue is consistently reproducible.

**Comment by Dr-Denzy** [comment]
@charles2588 could you try to see if you could reproduce this issue with Airflow 2.0.2

**Comment by kaxil** [comment]
Please provide reproduction steps and output of `pip freeze` please

**Comment by charles2588** [comment]
@kaxil 
For Airflow 2.0

```
pip freeze
alembic==1.4.3
amqp==2.6.1
apache-airflow==2.0.0
apache-airflow-providers-celery==1.0.1
apache-airflow-providers-ftp==1.0.1
apache-airflow-providers-http==1.1.1
apache-airflow-providers-imap==1.0.1
apache-airflow-providers-postgres==1.0.1
apache-airflow-providers-redis==1.0.1
apache-airflow-providers-sqlite==1.0.2
apispec==3.3.2
argcomplete==1.12.2
attrs==20.3.0
Babel==2.9.0
billiard==3.6.4.0
cached-property==1.5.2
cattrs==1.1.1
celery==4.4.7
certifi==2020.11.8
cffi==1.14.4
chardet==3.0.4
click==7.1.2
clickclick==20.10.2
colorama==0.4.4
colorlog==4.0.2
commonmark==0.9.1
connexion==2.7.0
croniter==0.3.36
cryptography==3.2.1
defusedxml==0.6.0
dill==0.3.3
dnspython==1.16.0
docutils==0.16
elasticsearch==7.10.1
email-validator==1.1.2
Flask==1.1.2
Flask-AppBuilder==3.1.1
Flask-Babel==1.0.0
Flask-Caching==1.9.0
Flask-JWT-Extended==3.25.0
Flask-Login==0.4.1
Flask-OpenID==1.2.5
Flask-SQLAlchemy==2.4.4
flask-swagger==0.2.13
Flask-WTF==0.14.3
flower==0.9.7
funcsigs==1.0.2
graphviz==0.15
gunicorn==19.10.0
humanize==3.4.1
idna==2.10
importlib-metadata==1.7.0
importlib-resources==1.5.0
inflection==0.5.1
iso8601==0.1.13
itsdangerous==1.1.0
Jinja2==2.11.2
json-merge-patch==0.2
jsonschema==3.2.0
kombu==4.6.11
lazy-object-proxy==1.4.3
lockfile==0.12.2
Mako==1.1.3
Markdown==3.3.3
MarkupSafe==1.1.1
marshmallow==3.9.1
marshmallow-enum==1.5.1
marshmallow-oneofschema==2.1.0
marshmallow-sqlalchemy==0.23.1
natsort==7.1.0
numpy==1.19.4
openapi-spec-validator==0.2.9
pandas==1.1.4
pendulum==2.1.2
prison==0.1.3
prometheus-client==0.8.0
psutil==5.7.3
psycopg2-binary==2.8.6
pycparser==2.20
Pygments==2.7.2
PyJWT==1.7.1
pyrsistent==0.17.3
python-daemon==2.2.4
python-dateutil==2.8.1
python-editor==1.0.4
python-nvd3==0.15.0
python-slugify==4.0.1
python3-openid==3.2.0
pytz==2020.4
pytzdata==2020.1
PyYAML==5.3.1
redis==3.5.3
requests==2.23.0
rich==9.2.0
setproctitle==1.2
six==1.15.0
SQLAlchemy==1.3.20
SQLAlchemy-JSONField==1.0.0
SQLAlchemy-Utils==0.36.8
swagger-ui-bundle==0.0.8
tabulate==0.8.7
tenacity==6.2.0
termcolor==1.1.0
text-unidecode==1.3
thrift==0.13.0
tornado==6.1
typing-extensions==3.7.4.3
tzlocal==1.5.1
unicodecsv==0.14.1
urllib3==1.25.11
vine==1.3.0
Werkzeug==1.0.1
WTForms==2.3.3
zipp==3.4.0
```

Repro steps:-
1. Just create a new DAG with 1 bashoperator.
2. Do not trigger it, just go to Graph View
3. Open the Task instance node and then click Instance details button.
In Airflow 2.0 , this results in error consistently.

that "2021-04-16T23%3A01%3A08.044070" that float part in the url causes this error which happens when you do not trigger it.

http://localhost:8080/task?dag_id=xcom_dag&task_id=downloading_data&execution_date=2021-04-16T23%3A01%3A08.044070%2B00%3A00


@Dr-Denzy I cannot repro above with Airflow 2.0.2.

**Comment by Dr-Denzy** [comment]
I could not reproduce the issue in both 2.0.0 and 2.0.2. And since you got it working with 2.0.2. Then we can close this issue. @kaxil

**Comment by kaxil** [comment]
Oh yeah -- this was most likely fixed in 2.0.1 itself. Thanks for testing @Dr-Denzy and @charles2588 for confirming that 2.0.2 does not suffer from this issue.
